crowbar_framework/app/models/deployer_service.rb

```diff
@@ -82,6 +82,7 @@ def transition(inst, name, state)
       unless node.crowbar["crowbar"]["pending"].nil?
         roles.concat(node.crowbar["crowbar"]["pending"].values)
       end
+      roles << node.run_list_to_roles
```

Comment:
there's a ```roles.flatten!``` line after this one in the original PR. We may need that one (still investigating it).